platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultConfigurationCache.kt

```diff
@@ -452,17 +452,19 @@ class DefaultConfigurationCache internal constructor(
     fun checkFingerprint(): CheckedFingerprint = buildOperationRunner.withFingerprintCheckOperations {
         val candidates = loadCandidateEntries()
         val result = searchForValidEntry(candidates)
-        if (result is CheckedFingerprint.Valid) {
-            updateMostRecentEntry(result.entryId)
+        val checkedFingerprint = result.second
+        if (checkedFingerprint is CheckedFingerprint.Valid) {
+            updateMostRecentEntry(checkedFingerprint.entryId)
         }
         result
     }
 
     private
-    fun searchForValidEntry(candidates: List<CandidateEntry>): CheckedFingerprint {
-        var firstInvalidResult: CheckedFingerprint? = null
+    fun searchForValidEntry(candidates: List<CandidateEntry>): Pair<String?, CheckedFingerprint> {
```

Comment:
💅 Because the first element is a `String?` I wouldn't object to either introducing a local `data class EntrySearchResult(val originBuildInvocationId: String?, val result: CheckedFingerprint)` or a `typealias OriginBuildInvocationId = String` to make these type signatures more meaningful

Reply:
A added a data class.